modules/sdk-core/src/bitgo/tss/common.ts

```diff
@@ -68,3 +70,51 @@ export async function sendSignatureShare(
     })
     .result();
 }
+
+/**
+ * Verifies that a TSS wallet signature was produced with the expected key and that the signed data contains the
+ * expected common keychain as well as the expected user and backup key ids
+ */
+export async function commonVerifyWalletSignature(params: {
+  walletSignature: openpgp.Key;
+  bitgoPub: openpgp.Key;
+  commonKeychain: string;
+  userKeyId: string;
+  backupKeyId: string;
+}): Promise<{ value: ArrayBuffer }[]> {
+  const { walletSignature, bitgoPub, commonKeychain, userKeyId, backupKeyId } = params;
+
+  // The result of verifyPrimaryUser contains two items:
```

Comment:
My apologies for providing incorrect info here.    `.verifyPrimaryUser` doesn't interact with subkeys.   It only checks certifications of the main key.

the code you have below:
```
const verificationResult = await walletSignature.verifyPrimaryUser([bitgoPub]);
const isValid = verificationResult.some((result) => result.valid);
```
is fine.   As this will check that the wallet pgp pubkey has at least ONE of it's attached certifications from the bitgo hsm pubkey (which is all we care about).   I mistakenly led you astray by thinking we were dealing with u-value subkeys, which we are not.  sorry about that.   This is only dealing with pgp CA keys certifying other pgp CA keys, and attaching notifcation data.

Reply:
I would remove the comment here regarding subkeys, as that doesnt apply here

Reply:
Comment is removed.